github/resource_github_repository_test.go

```diff
@@ -813,6 +813,52 @@ func TestAccGithubRepositoryPages(t *testing.T) {
 
 	})
 
+	t.Run("manages private pages feature for a repository", func(t *testing.T) {
+		config := fmt.Sprintf(`
+			resource "github_repository" "test" {
+				name         = "tf-acc-%s"
+				auto_init    = true
+				pages {
+					source {
+						branch = "main"
+					}
+					public = false
+				}
+			}
+		`, randomID)
+
+		check := resource.ComposeTestCheckFunc(
```

Comment:
Thank you for adding a test to capture this behavior! Is this check passing for you? When I run it, I'm getting `Check failed: Check 1/1 error: github_repository.test: Attribute 'pages.0.public' expected "false", got "true"`. 

It looks like we're only getting the "public" option, never setting it anywhere. Wouldn't we need some logic in the Create and Update functions to set this attribute?